databricks/koalas/indexes.py

```diff
@@ -579,6 +579,8 @@ def names(self) -> List[Union[Any, Tuple]]:
     def names(self, names: List[Union[Any, Tuple]]) -> None:
         if not is_list_like(names):
             raise ValueError("Names must be a list-like")
+        if not is_hashable(names[0]):
```

Comment:
Why only `names[0]`?

Reply:
Oh, sorry it was my mistake.

Btw, I think we should check this in `__new__` since pandas < 1.0.0 doesn't raise error even when name is not hashable.

```python
# pandas < 1.0.0 doesn't raise Exception for this case
>>> pd.Index([1, 2, 3], name=["0", "1"])
Int64Index([1, 2, 3], dtype='int64', name=['0', '1'])

# whereas pandas >= 1.0.0 raises Exception
>>> pd.Index([1, 2, 3], name=["0", "1"])
Traceback (most recent call last):
...
TypeError: Index.name must be a hashable type
```

So, if underlying pandas is lower than 1.0.0, it raises `AssertionError` rather than `TypeError` in `__new__`.

```python
>>> ks.Index([1, 2, 3], name=["0", "1"])
Traceback (most recent call last):
...
AssertionError: OrderedDict([('__index_level_0__', (['0', '1'],))])
```